File: _includes/app/basic-sample.rb
# Import the driver.
require 'pg'

# Connect to the cluster.
conn = PG.connect(user: 'maxroach', dbname: 'bank', host: 'localhost', port: 26257)

# Insert two rows into the table.
conn.exec("INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 250)")

# Check account balances.
puts "Initial account balances:"
conn.exec("SELECT id, balance FROM accounts") do |res|
        res.each do |row|
                puts row
        end
end

# Close communication with the database.
conn.close()

File: _includes/app/basic-sample.js
var async = require('async');

// Require the driver.
var pg = require('pg');

// Connect to the cluster.
var config = {
  user: 'maxroach',
  host: 'localhost',
  database: 'bank',
  port: 26257
};

pg.connect(config, function (err, client, done) {
  if (err) {
    return console.error('could not connect to cockroachdb', err);
  }
  async.waterfall([
    function (next) {
      // Insert two rows into the table.
      client.query("INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 250);", next);
    },
    function (results, next) {
      // Check account balances.
      client.query('SELECT id, balance FROM accounts;', next);
    },
  ],
  function (err, results) {
    if (err) {
      return console.error('error inserting into and selecting from accounts', err);
    }

    console.log('Initial account balances:');
    results.rows.forEach(function (row) {
      console.log(row);
    });

    // Close communication with the database and exit.
    done();
    process.exit();
  });
});
